security/https-2way-authz/src/test/java/io/quarkus/ts/openshift/security/https/twoway/authz/SecurityHttps2wayAuthzTest.java

```diff
@@ -0,0 +1,158 @@
+package io.quarkus.ts.openshift.security.https.twoway.authz;
+
+import io.quarkus.test.common.http.TestHTTPResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.ts.openshift.common.DisabledOnQuarkus;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+import org.junit.jupiter.api.Test;
+
+import javax.net.ssl.SSLContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import static io.quarkus.ts.openshift.common.util.HttpsAssertions.assertTls13OnlyHandshakeError;
+import static io.quarkus.ts.openshift.common.util.HttpsAssertions.assertTlsHandshakeError;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@QuarkusTest
+@DisabledOnQuarkus(version = "1\\.[345]\\..*", reason = "https://github.com/quarkusio/quarkus/pull/8991")
+public class SecurityHttps2wayAuthzTest {
+    // not using RestAssured because we want 100% control over certificate & hostname verification
+
+    private static final char[] CLIENT_PASSWORD = "client-password".toCharArray();
+
+    @TestHTTPResource(value = "/", ssl = true)
+    private String url;
+
+    @TestHTTPResource(value = "/secured", ssl = true)
+    private String urlWithAuthz;
+
+    @TestHTTPResource(value = "/")
+    private String insecureUrl;
+
+    @Test
+    public void https_authenticatedAndAuthorizedClient() throws IOException, GeneralSecurityException {
+        SSLContext sslContext = SSLContexts.custom()
+                .setKeyStoreType("pkcs12")
+                .loadKeyMaterial(new File("target/client-keystore.pkcs12"), CLIENT_PASSWORD, CLIENT_PASSWORD)
+                .loadTrustMaterial(new File("target/client-truststore.pkcs12"), CLIENT_PASSWORD)
+                .build();
+        try (CloseableHttpClient httpClient = HttpClients.custom()
+                .setSSLContext(sslContext)
+                .setSSLHostnameVerifier(new DefaultHostnameVerifier())
+                .build()) {
+
+            Executor executor = Executor.newInstance(httpClient);
+
+            {
```

Comment:
Why is this `{ ... }` approach needed?

Reply:
It's not needed, but I believe it improves readability. There are 2 independent requests, so I have 2 independent blocks of code. If I didn't, I'd have to either reuse the `response` variable, or have a different name for the 2nd one, and I find both approaches more error-prone than this.

Reply:
ok, thanks for explaining